js/iframe-coordinator/src/messages/specs/ClientToHost.spec.ts

```diff
@@ -0,0 +1,237 @@
+import { ClientToHost, validate } from '../ClientToHost';
+import { LabeledNavRequest } from '../NavRequest';
+import { LabeledToast } from '../Toast';
+
+describe('ClientToHost', () => {
+  describe('validating an invalid message type', () => {
+    const testMessage = {
+      msgType: 'foobar',
+      msg: 'test-data'
+    };
+
+    let testResult: ClientToHost | null;
+    beforeEach(() => {
+      testResult = validate(testMessage);
+    });
+    it('should return a null message', () => {
+      expect(testResult).toBeNull();
+    });
+  });
+
+  describe('validating publish type', () => {
+    describe('when payload is a string', () => {
+      const testMessage = {
+        msgType: 'publish',
+        msg: {
+          topic: 'test.topic',
+          payload: 'test.payload'
+        }
+      };
+      let testResult: ClientToHost | null;
+      beforeEach(() => {
+        testResult = validate(testMessage);
+      });
+      it('should return the validated message', () => {
+        expect(testResult).toEqual(testMessage);
+      });
+    });
+
+    describe('when payload is an object', () => {
+      const testMessage = {
+        msgType: 'publish',
+        msg: {
+          topic: 'test.topic',
+          payload: { testData: 'test.data' }
+        }
+      };
+      let testResult: ClientToHost | null;
+      beforeEach(() => {
+        testResult = validate(testMessage);
+      });
+      it('should return the validated message', () => {
+        expect(testResult).toEqual(testMessage);
+      });
+    });
+
+    describe('when topic is missing', () => {
+      const testMessage = {
+        msgType: 'publish',
+        msg: {
+          payload: { testData: 'test.data' }
+        }
+      };
+      let testResult: ClientToHost | null;
+      beforeEach(() => {
+        testResult = validate(testMessage);
+      });
+      it('should return the validated message', () => {
+        expect(testResult).toBeNull();
+      });
+    });
+
+    describe('when payload is missing', () => {
+      const testMessage = {
+        msgType: 'publish',
+        msg: {
+          topic: 'test.topic'
+        }
+      };
+      const expectedMessage = {
+        msgType: 'publish',
+        msg: {
+          topic: 'test.topic',
+          payload: undefined
```

Comment:
This is the other annoying thing.  The decoder adds a key for optional properties even if they weren't in the original json.  They are still undefined, so its not a huge deal, but its still not expected